substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -149,45 +304,594 @@ public Path filePath() {
         public Path cachePath() {
             return cachePath;
         }
+    }
+
+    private abstract class NativeImageDebugTypeInfo extends NativeImageDebugFileInfo implements DebugTypeInfo {
+        protected final HostedType hostedType;
+
+        @SuppressWarnings("try")
+        protected NativeImageDebugTypeInfo(HostedType hostedType) {
+            super(hostedType);
+            this.hostedType = hostedType;
+        }
+
+        @SuppressWarnings("try")
+        @Override
+        public void debugContext(Consumer<DebugContext> action) {
+            try (DebugContext.Scope s = debugContext.scope("DebugTypeInfo", typeName())) {
+                action.accept(debugContext);
+            } catch (Throwable e) {
+                throw debugContext.handle(e);
+            }
+        }
+
+        public String toJavaName(@SuppressWarnings("hiding") HostedType hostedType) {
+            return getJavaType(hostedType, true).toJavaName();
+        }
+
+        @Override
+        public String typeName() {
+            return toJavaName(hostedType);
+        }
+
+        @Override
+        public int size() {
+            if (hostedType instanceof HostedInstanceClass) {
+                // We know the actual instance size in bytes.
```

Comment:
`//` comments should be avoided. `mx eclipseformat` can behave badly when code gets refactored that uses `//` comments.

Reply:
Done